File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/items/media/files/menu/AbstractFileListItemNowPlayingHandler.java
package com.lasthopesoftware.bluewater.client.browsing.items.media.files.menu;

import android.content.BroadcastReceiver;
import android.content.IntentFilter;
import android.view.View;
import android.widget.RelativeLayout;

import androidx.localbroadcastmanager.content.LocalBroadcastManager;

import com.lasthopesoftware.bluewater.client.playback.service.broadcasters.PlaylistEvents;

/**
 * Created by david on 4/14/15.
 */
public abstract class AbstractFileListItemNowPlayingHandler extends BroadcastReceiver {

    private final RelativeLayout fileTextViewContainer;

    private final View.OnAttachStateChangeListener onAttachStateChangeListener;

    private final LocalBroadcastManager localBroadcastManager;

    public AbstractFileListItemNowPlayingHandler(FileListItemContainer fileListItem) {
        fileTextViewContainer = fileListItem.getTextViewContainer();

        localBroadcastManager = LocalBroadcastManager.getInstance(fileTextViewContainer.getContext());

        localBroadcastManager.registerReceiver(this, new IntentFilter(PlaylistEvents.onPlaylistChange));

        onAttachStateChangeListener = new View.OnAttachStateChangeListener() {

            @Override
            public void onViewDetachedFromWindow(View v) {
                localBroadcastManager.unregisterReceiver(AbstractFileListItemNowPlayingHandler.this);
            }

            @Override
            public void onViewAttachedToWindow(View v) { }
        };

        fileTextViewContainer.addOnAttachStateChangeListener(onAttachStateChangeListener);
    }

    public void release() {
        localBroadcastManager.unregisterReceiver(this);
        fileTextViewContainer.removeOnAttachStateChangeListener(onAttachStateChangeListener);
    }
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/items/media/files/menu/AbstractFileListItemNowPlayingHandler.kt
package com.lasthopesoftware.bluewater.client.browsing.items.media.files.menu

import android.content.BroadcastReceiver
import android.content.IntentFilter
import android.view.View
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import com.lasthopesoftware.bluewater.client.playback.service.broadcasters.PlaylistEvents

abstract class AbstractFileListItemNowPlayingHandler(fileListItem: FileListItemContainer) : BroadcastReceiver(), View.OnAttachStateChangeListener {
	private val fileTextViewContainer = fileListItem.textViewContainer
	private val localBroadcastManager = LocalBroadcastManager.getInstance(fileTextViewContainer.context)

	init {
		localBroadcastManager.registerReceiver(this, IntentFilter(PlaylistEvents.onPlaylistTrackChange))
		fileTextViewContainer.addOnAttachStateChangeListener(this)
	}

	fun release() {
		localBroadcastManager.unregisterReceiver(this)
		fileTextViewContainer.removeOnAttachStateChangeListener(this)
	}

	override fun onViewDetachedFromWindow(v: View) {
		localBroadcastManager.unregisterReceiver(this)
	}

	override fun onViewAttachedToWindow(v: View) {}
}
